File: 4.8-bc/ts/stage-3-hexagonal/src/commerce/module.ts
// Self-wiring entry point для Commerce BC.
import type { DbPool } from "../shared/db.js";
import type { EventBus } from "../shared/events/bus.js";

import { Service } from "./app/service.js";
import { Handler } from "./infra/http/handler.js";
import { PgOrderRepo } from "./infra/postgres/orderRepo.js";

export interface Module {
  service: Service;
  handler: Handler;
}

export function newModule(db: DbPool, bus: EventBus): Module {
  const repo = new PgOrderRepo(db);
  const service = new Service(repo, bus);
  const handler = new Handler(service);
  return { service, handler };
}

File: 4.8-bc/py/stage-3-hexagonal/app/commerce/module.py
"""Commerce BC self-wiring."""
from __future__ import annotations

from dataclasses import dataclass

from fastapi import APIRouter
from sqlalchemy.ext.asyncio import async_sessionmaker

from app.commerce.app.service import Service
from app.commerce.infra.http.handler import build_router
from app.commerce.infra.postgres.order_repo import OrderRepo
from app.shared.events import EventBus


@dataclass
class Module:
    service: Service
    _router: APIRouter

    def routes(self) -> APIRouter:
        return self._router


def New(session_maker: async_sessionmaker, bus: EventBus) -> Module:
    repo = OrderRepo(session_maker)
    svc = Service(repo, bus)
    router = build_router(svc)
    return Module(service=svc, _router=router)
